### ISSUE TITLE
Styled-components break with react-router v4

### ISSUE THREAD

**Issue by gbozee** [styled-components/styled-components#75]
Trying to make use of a styled-component with react-router v4 breaks with the following error

    StyledComponent.js:71 Uncaught TypeError: subscribe is not a function
    componentWillMount @ StyledComponent.js:71

Confirmed that this isn't an issue with react-router because
This works

    const Li = props => <li style={{display: "inline-block"}} {...props}/>

But this fails and throws the above error when used with react-router v4

    const Li = styled.li`
    display: inline-block`

**Comment by madeinfree** [comment]
I got the same error.
I found this issue happened [StyledComponent.js#L44](https://github.com/styled-components/styled-components/blob/master/src/models/StyledComponent.js#L44), it can get the `this.context` but didn't have `__styled-components__`, I try to fixed this line become 
```
this.unsubscribe = subscribe && subscribe(function (theme) {
  // This will be called once immediately
  _this3.setState({ theme: theme });
  });
```
it works fine

**Comment by mxstbr** [comment]
Interesting, @madeinfree would you mind submitting a PR with this change? Then I can take a look if that fixes the issue once and for all! Thanks!

**Comment by madeinfree** [comment]
@mxstbr no problem ~

**Comment by sheepsteak** [comment]
I think this may be because `react-router` also uses broadcasts via [react-broadcast](https://github.com/ReactTraining/react-broadcast) and there is a clash as both put `broadcasts` onto the context. I _think_ the fix above is just papering over the cracks.

Maybe it would be better to remove the broadcast stuff in this module and depend on `react-broadcast` instead? The code is very similar and there would be no overwriting happening on the context?

**Comment by mxstbr** [comment]
> I think this may be because react-router also uses broadcasts via react-broadcast and there is a clash as both put broadcasts onto the context. I think the fix above is just papering over the cracks.

Interesting theory. We could just not use the `broadcast` context property and use `theme` instead?

> Maybe it would be better to remove the broadcast stuff in this module and depend on react-broadcast instead? The code is very similar and there would be no overwriting happening on the context.

The reason we don't do this (if you look at the PR I actually initially built it this way: #56) is that it would mean that every styled component is wrapped in _two_ components instead of one, which has a non-negligible performance impact. I thought I integrated it properly so it doesn't matter if there's another broadcast happening, but obviously I was wrong.

**Comment by garmjs** [comment]
I'm getting the same error with react-router v4, i think i'm goin to use @madeinfree fix for now

**Comment by mxstbr** [comment]
Can replicate – working on a fix now! 🔥

**Comment by mxstbr** [comment]
Could you try out #86 in your project and report back if it fixes the issue? It did for me in a simple test, but I want to make 100% sure!

**Comment by madeinfree** [comment]
I can try ~~ 👍🏻

**Comment by gbozee** [comment]
Just tested the PR. Issue resolved.

**Comment by mxstbr** [comment]
Awesomeee, thanks!

**Comment by mxstbr** [comment]
Released as `v1.0.4` – thanks for reporting and testing this everyone!
